exploding-kittens-frontend/src/app/RoomList.tsx

```diff
@@ -0,0 +1,44 @@
+'use client'
+import { useState, useEffect } from 'react'
+import Link from 'next/link'
+
+export const RoomList = () => {
+  const [rooms, setRooms] = useState<Room[]>([])
+
+  useEffect(() => {
+    const findRooms = async () => {
+      await fetch('http://localhost:3000/rooms', {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      })
+      .then(res => res.json())
+      .then((data: Room[]) => {
+        console.log(data)
+        setRooms(data)
+      })
+    }
+    if (rooms.length < 1) findRooms()
+  }, [])
```

Comment:
Not needed now but, this could be converted to a server component since it's all static data(it's a little faster to load server components I think, and doesn't have white flash that could happen for complex client components)

```suggestion
import Link from 'next/link'

export const RoomList = () => {


    const findRooms = async () => {
      return await fetch('http://localhost:3000/rooms', {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        }
        //nextjs will auto cache,this stops it
        cache:'no-store'
      })
      .then(res => res.json())
      .then((data: Room[]) => {
        console.log(data)
        return data
      })
    }
    const rooms = await findRooms()
```